pvlib/ivtools.py

```diff
@@ -262,6 +264,177 @@ def fit_sde_sandia(voltage, current, v_oc=None, i_sc=None, v_mp_i_mp=None,
+        process.
+    beta_voc: numeric
+        The open-circuit voltage (v_oc) temperature coefficient of the
+        module in units of %/K. It is converted in V/K for the computing
+        process.
+    cells_in_series: numeric
+        Number of cell in the module.
+        Optional input, but helps to insure the convergence of the computing.
+    temp_ref: numeric, default 25
+        Reference temperature condition [C]
+    irrad_ref: numeric, default 1000
+        Reference irradiance condition [W/m2]
+
+    Returns
+    -------
+    Dictionnary with the following elements:
+
+    * 'I_L_ref': numeric
+        Light-generated current in amperes at std conditions.
+    * 'I_o_ref': numeric
+        Diode saturation curent in amperes at std conditions
+    * 'R_s': numeric
+        Series resistance in ohms. Note that '_ref' is not mentionned
+        in the name because this resistance is not sensible to the
+        conditions of test.
+    * 'R_sh_ref': numeric
+        Shunt resistance in ohms at std conditions.
+    * 'a_ref' : numeric
+        Modified ideality factor at std conditions.
+        The product of the usual diode ideality factor (n, unitless),
+        number of cells in series (Ns), and cell thermal voltage at
+        specified effective irradiance and cell temperature.
+    * 'alpha_sc': numeric
+        Caution!: Different from the input because of the unit.
+        The short-circuit current (i_sc) temperature coefficient of the
+        module in units of A/K.
+    * 'EgRef': numeric
+        Energy of bandgap of semi-conductor used (depending on celltype) [J]
+    * 'dEgdT': numeric
+        Variation of bandgap according to temperature [J/K]
+    * 'irrad_ref': numeric
+        Reference irradiance condition [W/m2]
+    * 'temp_ref': numeric
+        Reference temperature condition [C]
+
+    References
+    ----------
+    [1] W. De Soto et al., "Improvement and validation of a model for
+    photovoltaic array performance", Solar Energy, vol 80, pp. 78-88,
+    2006.
+
+    [2] John A Dufﬁe ,William A Beckman, "Solar Engineering of Thermal
+    Processes", Wiley, 2013
+    """
+    # Constants
+    k = constants.Boltzmann  # in J/K, or 8.617e-5 eV/K
+    q = constants.elementary_charge  # in J/V, or 1 eV
+
+    Tref = temp_ref + 273.15  # in K
+
+    if celltype.lower() in ['monosi', 'polysi', 'multisi',
+                            'mono-c-si', 'multi-c-si']:
+        dEgdT = -0.0002677  # valid for silicon
+        EgRef = 1.796e-19  # in J, valid for silicon
+    elif celltype.lower() in ['cis', 'cigs', 'cdte', 'amorphous', 'thin film']:
+        raise NotImplementedError
+    else:
+        raise ValueError('Unknown cell type.')
+
+    # Conversion from %/K to A/K & V/K
+    alpha_sc = alpha_sc*i_sc/100
+    beta_voc = beta_voc*v_oc/100
+
+    def pv_fct(params, specs):
+        """Returns the system of equations used for computing the
+        single-diode 5 parameters.
```

Comment:
"Evaluates the systems of equations used to solve for the single diode equation parameters."